src/util/path/path.ts

```diff
@@ -754,81 +881,78 @@ export const getPointOnPath = (
  * ];
  *
  */
-export const parsePath = (pathString) => {
-  // one of commands (m,M,l,L,q,Q,c,C,etc.) followed by non-command characters (i.e. command values)
-  const re = rePathCommand,
-    rNumber = '[-+]?(?:\\d*\\.\\d+|\\d+\\.?)(?:[eE][-+]?\\d+)?\\s*',
-    rNumberCommaWsp = `(${rNumber})${commaWsp}`,
-    rFlagCommaWsp = `([01])${commaWsp}?`,
-    rArcSeq = `${rNumberCommaWsp}?${rNumberCommaWsp}?${rNumberCommaWsp}${rFlagCommaWsp}${rFlagCommaWsp}${rNumberCommaWsp}?(${rNumber})`,
-    regArcArgumentSequence = new RegExp(rArcSeq, 'g'),
-    result = [];
-
-  if (!pathString || !pathString.match) {
-    return result;
-  }
-  const path = pathString.match(/[mzlhvcsqta][^mzlhvcsqta]*/gi);
-
-  for (let i = 0, len = path.length; i < len; i++) {
-    const currentPath = path[i];
-    const coordsStr = currentPath.slice(1).trim();
-    const coords = [];
-    let command = currentPath.charAt(0);
-    const coordsParsed = [command];
-
-    if (command.toLowerCase() === 'a') {
-      // arcs have special flags that apparently don't require spaces so handle special
-      for (let args; (args = regArcArgumentSequence.exec(coordsStr)); ) {
-        for (let j = 1; j < args.length; j++) {
-          coords.push(args[j]);
-        }
-      }
-    } else {
-      let match;
-      while ((match = re.exec(coordsStr))) {
-        coords.push(match[0]);
-      }
-    }
+export const parsePath = (pathString: string): TComplexPathData => {
+  // clean the string
+  // add spaces around the numbers
+  pathString = pathString.replace(
+    new RegExp(`(${numberRegExStr})`, 'gi'),
+    ' $1 '
+  );
+  // replace annoying commas and arbitrary whitespace with single spaces
+  pathString = pathString.replace(/,/gi, ' ');
+  pathString = pathString.replace(/\s+/gi, ' ');
```

Comment:
```suggestion
  pathString = pathString.replace(
    new RegExp(`(${numberRegExStr})`, 'gi'),
    ' $1 '
  )
  // replace annoying commas and arbitrary whitespace with single spaces
  .replace(/,/gi, ' ')
  .replace(/\s+/gi, ' ');
```

would this work?

Reply:
I think so but I'll run the tests really quick and check